lib/mv_opentelemetry/tesla.ex

```diff
@@ -0,0 +1,91 @@
+defmodule MvOpentelemetry.Tesla do
+  @moduledoc false
+
+  use MvOpentelemetry.SpanTracer,
+    name: :tesla,
+    events: [
+      [:tesla, :request, :start],
+      [:tesla, :request, :stop],
+      [:tesla, :request, :exception]
+    ]
+
+  require OpenTelemetry.SemanticConventions.Trace, as: Trace
+
+  def handle_event([:tesla, :request, :start], _measurements, meta, opts) do
+    %{
+      env: %Tesla.Env{
```

Comment:
I would recommend not matching on the struct type (Tesla.Env) and instead treat it like a map to avoid adding some compile time dependency.